### PR TITLE
Channels 1088: refactor engage folder structure

### PR DESCRIPTION
**Pull request by alfrimpong**
[JIRA](https://segment.atlassian.net/browse/CHANNELS-1088)

This PR moves shared code into the actions-shared package and changes the folder structure of engage destinations to match other destinations.

## Testing

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [x] [Segmenters] Tested in the staging environment

### CODE REVIEW

**Comment by alfrimpong** [comment]
I recommend reviewing this PR commit-by-commit. It is far less daunting that way. Unfortunately, splitting this PR up would take weeks due to the deployment process.

**Comment by alfrimpong** [comment]
Dont be afraid. 
51 of the 73 files in this PR were moved or deleted. 
There were no changes to business logic. Most of the changes involved imports

**Comment by cogwizzle** [comment]
I do not see any thing glaring in this PR. It is rather large. I would recommend getting a second set of eyes on it for sure.

**Review by cogwizzle — DISMISSED** [review]

**Review by pmunin — DISMISSED** [review]
thanks for doing it, looks good to me

**Review by joe-ayoub-segment — DISMISSED** [review]

**Comment by joe-ayoub-segment** [comment]
hi @alfrimpong - just back from PTO today - can you try rebasing from main to see if it fixes the 2 failing CI checks please?

**Comment by alfrimpong** [comment]
> hi @alfrimpong - just back from PTO today - can you try rebasing from main to see if it fixes the 2 failing CI checks please?

Moving the pr back to draft to deal with the CI issue and do more testing.

File: packages/browser-destinations/destinations/userpilot/package.json [hunk 1]
```diff
@@ -15,8 +15,8 @@
   },
   "typings": "./dist/esm",
   "dependencies": {
-    "@segment/actions-core": "^3.109.0",
-    "@segment/browser-destination-runtime": "^1.39.0"
+    "@segment/actions-core": "^3.110.0",
+    "@segment/browser-destination-runtime": "^1.40.0"
```
[the remark this answers is not in the document]
Comment (joe-ayoub-segment) [reply packages/browser-destinations/destinations/userpilot/package.json]: Also "@segment/actions-core": "^3.110.0" is up to "^3.111.0" on the main branch.

File: packages/core/package.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "@segment/actions-core",
   "description": "Core runtime for Destinations Actions.",
-  "version": "3.109.0",
+  "version": "3.110.0",
```
Comment (joe-ayoub-segment) [inline packages/core/package.json]: hi @alfrimpong - "version": "3.110.0" is already up to "version": "3.111.0" on the main branch.

**Comment by alfrimpong** [comment]
@joe-ayoub-segment can we please get this considered for deployment in the next release?

**Review by joe-ayoub-segment — APPROVED** [review]

**Comment by joe-ayoub-segment** [comment]
Hi @alfrimpong looks good to me. 
Has this been tested or sanity checked somewhere? In Staging for example?

**Comment by alfrimpong** [comment]
@joe-ayoub-segment yes, it was deployed and tested on stage weeks ago.

**Review by joe-ayoub-segment — DISMISSED** [review]

**Comment by joe-ayoub-segment** [comment]
hi @alfrimpong . This PR has been deployed. Please check that it works as expected :)

**Comment by alfrimpong** [comment]
> hi @alfrimpong . This PR has been deployed. Please check that it works as expected :)

yep still works

**Pull request merged**